templates/chrony.conf.epp

```diff
@@ -1,12 +1,24 @@
 # NTP servers
-<% $chrony::servers.each |$server| { -%>
-server <%= $server.flatten.join(' ') %>
+<% $servers.keys.sort.each |$server| { -%>
+<%   if $servers[$server] and !$servers[$server].empty { -%>
+server <%= $server %> <%= $servers[$server].join(' ') %>
+<%   } else { -%>
+server <%= $server %>
 <% } -%>
-<% $chrony::pools.each |$pool| { -%>
-pool <%= $pool.flatten.join(' ') %>
+<%   } -%>
+<% $pools.keys.sort.each |$pool| { -%>
+<%   if $pools[$pool] and !$pools[$pool].empty { -%>
+pool <%= $pool %> <%= $pools[$pool].join(' ') %>
+<%   } else { -%>
+pool <%= $pool %>
+<%   } -%>
 <% } -%>
-<% $chrony::peers.each |$peer| { -%>
-peer <%= $peer.flatten.join(' ') %>
+<% $peers.keys.sort.each |$peer| { -%>
+<%   if $peers[$peer] and !$peers[$peer].empty { -%>
```

Comment:
Is this the same?
```suggestion
<%   if !empty($peers[$peer]) { -%>
```

If so, I would probably reverse the conditions so it becomes `if empty() {} else {}`.

Reply:
I don't think you can call empty on `undef` though?  I'll double check.

Reply:
https://puppet.com/docs/puppet/6.4/function.html#empty states:
> For backwards compatibility with the stdlib function with the same name the following data types are also accepted by the function instead of raising an error. Using these is deprecated and will raise a warning:
>
>    Numeric - false is returned for all Numeric values.
>    Undef - true is returned for all Undef values.

So yes, you can call it but you probably shouldn't is the answer?

Reply:
I'm ok with a warning (and I've already made the change!). I don't think many users will encounter this and TBH, supporting `undef` is a corner case that's easily avoided by the user.  eg the hiera in https://github.com/voxpupuli/puppet-chrony/pull/94#issuecomment-662155344
could be changed to 
```yaml
chrony::servers:
  ntp1:
    - minpoll 7
  ntp2: []
```

Reply:
https://github.com/puppetlabs/puppet/pull/6776#issuecomment-754136646 ;)